orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationClient.java

```diff
@@ -144,6 +147,54 @@ public CompletionPostResponse executeRequest(@Nonnull final CompletionPostReques
     return executeRequest(postRequest);
   }
 
+  /**
+   * Perform a request to the orchestration service using a module configuration provided as JSON
+   * string. This can be useful when building a configuration in the AI Launchpad UI and exporting
+   * it as JSON. Furthermore, this allows for using features that are not yet supported natively by
+   * the API.
+   *
+   * <p><b>NOTE:</b> This method does not support streaming.
+   *
+   * @param prompt The input parameters and optionally message history to use for prompt execution.
+   * @param moduleConfig The module configuration in JSON format.
+   * @return The completion response.
+   * @throws OrchestrationClientException If the request fails.
+   */
+  @Beta
+  @Nonnull
+  public OrchestrationChatResponse executeRequestFromJsonModuleConfig(
+      @Nonnull final OrchestrationPrompt prompt, @Nonnull final String moduleConfig)
+      throws OrchestrationClientException {
+    if (!prompt.getMessages().isEmpty()) {
+      throw new IllegalArgumentException(
+          "Prompt must not contain any messages when using a JSON module configuration, as the template is already defined in the JSON.");
+    }
```

Comment:
An OrchestrationPrompt  is not the right object then. Why not accept a map of input params?

Reply:
Because that is missing the (optional) messages history. 

> OrchestrationPrompt is not the right object

I'd argue it is. We will have further cases where passing messages is not allowed, e.g. when using TemplateRef. If we want we could create another Prompt object for that to make this distinction, but so far I figured it's not worth it.